File: src/main/java/me/cephetir/skyskipped/features/impl/fragrun/PartyCommand.java
/*
 *
 * DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
 * Version 2, December 2004
 *
 * Copyright (C) 2021 Cephetir
 *
 * Everyone is permitted to copy and distribute verbatim or modified
 * copies of this license document, and changing it is allowed as long
 * as the name is changed.
 *
 * DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
 * TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION
 *
 *  0. You just DO WHAT THE FUCK YOU WANT TO.
 */

package me.cephetir.skyskipped.features.impl.fragrun;

import gg.essential.api.EssentialAPI;
import me.cephetir.skyskipped.SkySkipped;
import me.cephetir.skyskipped.config.Config;
import net.minecraft.client.Minecraft;
import net.minecraft.command.CommandBase;
import net.minecraft.command.ICommandSender;
import net.minecraft.util.BlockPos;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.TickEvent;

import java.util.Collections;
import java.util.List;

public class PartyCommand extends CommandBase {
    private boolean started = false;

    @Override
    public String getCommandName() {
        return "fragrun";
    }

    @Override
    public List<String> getCommandAliases() {
        return Collections.singletonList("frag");
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return null;
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 0;
    }

    @Override
    public List<String> addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos) {
        return null;
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) {
        if (args.length == 0) {
            SkySkipped.features.getLeaveCommand().start(true);
        }
    }

    public void start() {
        if (started || !EssentialAPI.getMinecraftUtil().isHypixel()) return;
        started = true;
        MinecraftForge.EVENT_BUS.register(this);
    }

    private int step = 0;
    private boolean startedd = false;
    @SubscribeEvent
    public void onTick(TickEvent.ClientTickEvent event) {
        if(startedd) return;
        new Thread(() -> {
            startedd = true;
            switch (step) {
                case 0: {
                    Minecraft.getMinecraft().thePlayer.sendChatMessage("/p leave");
                    timer(200L);
                    break;
                }
                case 1: {
                    Minecraft.getMinecraft().thePlayer.sendChatMessage("/p " + Config.BotName);
                    MinecraftForge.EVENT_BUS.unregister(this);
                    started = false;
                    step = 0;
                    break;
                }
            }
            startedd = false;
        }).start();
    }

    private void timer(long millis) {
        try {
            Thread.sleep(millis);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        step++;
    }
}

File: src/main/kotlin/me/cephetir/skyskipped/commands/dungeonCommands/PartyCommand.kt
/*
 * SkySkipped - Hypixel Skyblock QOL mod
 * Copyright (C) 2023  Cephetir
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package me.cephetir.skyskipped.commands.dungeonCommands

import gg.essential.api.EssentialAPI
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import me.cephetir.bladecore.core.event.BladeEventBus
import me.cephetir.bladecore.utils.threading.BackgroundScope
import me.cephetir.skyskipped.config.Config
import me.cephetir.skyskipped.features.Features
import me.cephetir.skyskipped.utils.mc
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraftforge.fml.common.gameevent.TickEvent.ClientTickEvent

class PartyCommand : CommandBase() {
    private var started = false

    override fun getCommandName(): String {
        return "fragrun"
    }

    override fun getCommandAliases(): List<String> {
        return listOf("frag")
    }

    override fun getCommandUsage(sender: ICommandSender): String? {
        return null
    }

    override fun getRequiredPermissionLevel(): Int {
        return 0
    }

    override fun processCommand(sender: ICommandSender, args: Array<String>) {
        Features.leaveCommand.start(true)
    }

    fun start() {
        if (started || !EssentialAPI.getMinecraftUtil().isHypixel()) return
        started = true
        BladeEventBus.subscribe(this)
    }

    private var step = 0
    private var startedd = false

    @SubscribeEvent
    fun onTick(event: ClientTickEvent) {
        if (startedd) return
        val toStop = this
        BackgroundScope.launch {
            startedd = true
            when (step) {
                0 -> {
                    mc.thePlayer.sendChatMessage("/p leave")
                    delay(200L)
                    step++
                }

                1 -> {
                    mc.thePlayer.sendChatMessage("/p " + Config.BotName.value)
                    BladeEventBus.unsubscribe(toStop, true)
                    started = false
                    step = 0
                }
            }
            startedd = false
        }
    }
}
